example_mvc/Controllers/RecipesController.cs

```diff
@@ -54,6 +56,31 @@ public async Task<IActionResult> Index(string searchString, string SortRecipe)
                     break;
 
             }
+            switch (CheckTag)
+            {
+                case "All":
+                    Recipe = from r in _context.Recipe
+                             select r;
+                    break;
+                case "Soup":
+                    Recipe = Recipe.Where(s => s.Soup == true);
+                    break;
+                case "Dinner":
+                    Recipe = Recipe.Where(s => s.Dinner == true);
+                    break;
+                case "Dessert":
+                    Recipe = Recipe.Where(s => s.Dessert == true);
+                    break;
+                case "Drink":
+                    Recipe = Recipe.Where(s => s.Drink == true);
+                    break;
+                case "Breakfast":
+                    Recipe = Recipe.Where(s => s.Breakfast == true);
+                    break;
+                case "Preserves":
+                    Recipe = Recipe.Where(s => s.Preserves == true);
+                    break;
```

Comment:
hardkodowanie tagow w ten sposob to malo elastyczne rozwiazanie :) Wybrazam sobie ze tag moze byc osobna encja a kazdy przepis moze miec przypisane 1..n tagow z uzyciem relacji wiele do wielu, zadzow to wytlumacze dokladnie o co chodzi